docs/architecture/_index.md

```diff
@@ -0,0 +1,188 @@
+
+When `viam-server` starts, it uses credentials stored locally to establish a connection with the Viam app over {{< glossary_tooltip term_id="webrtc" text="WebRTC" >}}.
+It fetches its configuration, which in our example contains the sensor pin information, from the Viam app.
+
+<details>
+  <summary>Click for information on local configuration</summary>
+  <p>
+  If you need to run your machine offline, you can manually create a local configuration file, eliminating the need for <code>viam-server</code> to fetch its config.<br><br>
+  The advantages of using the credentials and pulling the config from the Viam app are:
+  <ul>
+    <li>The config builder UI in the Viam app is more user-friendly than writing JSON manually</li>
+    <li>You can easily share configs across multiple identical machines and update them from one place</li>
+    <li>etc....TODO</li>
+  </ul>
+  If your machine usually runs offline but can connect online occasionally, it can pull a configuration from the Viam app and cache it locally to use until the next time it has an internet connection.
+  </p>
+</details><br>
+
+`viam-server` runs your machine, securely managing all communications between hardware and to the cloud.
+
+You can use the tools in the Viam app to remotely view sensor data as well as to change your machine's configuration, to view logs, and more.
+
+Now imagine you want to run code to send you an email when the temperature sensor reads over 100 degrees Fahrenheit:
+
+- You write code using any of the Viam [SDKs](/sdks/), for example the Viam Python SDK.
+- You then run this code either locally on the SBC, or on a separate computer such as a laptop.
+  Either way, you include a few lines of code at the top of your script that set up a connection to your machine's `viam-server` instance so you can access the sensor component and use the [sensor API](/components/sensor/#api) to get readings.
+
+  ![alt](/build/program/sdks/robot-client.png)
+
+Now, imagine you want to change to a different model of temperature sensor from a different brand:
+
+- You power down your device, disconnect the old sensor from your SBC and connect the new one.
+- You update your configuration in the Viam app to indicate what model you are using, and how it's connected (imagine this one uses USB instead of GPIO pins).
+- You turn your device back on, and `viam-server` automatically fetches the config updates.
+- You do not need to change your SDK code, because the API is the same for all models of sensor.
+
+## `viam-server`
+
+`viam-server` is the open-source executable binary that runs on your machine's SBC or other computer.
+
+`viam-server` can run your machine without any connection to the internet; it contains all the drivers for your hardware, and it manages local connections between hardware drivers, runs motion planning and vision services locally, and captures and stores data locally.
```

Comment:
I don't think saying it contains all the drivers is entirely true. Because modules aren't a part of viam-server